miio/g1vacuum.py

```diff
@@ -0,0 +1,281 @@
+
+_LOGGER = logging.getLogger(__name__)
+MIJIA_VACUUM_V2 = "mijia.vacuum.v2"
+
+MIOT_MAPPING = {
+    MIJIA_VACUUM_V2: {
+    # https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:vacuum:0000A006:mijia-v1:1
+    "battery": {"siid": 3, "piid": 1},
+    "charge_state": {"siid": 3, "piid": 2},
+    "error_code": {"siid": 2, "piid": 2},
+    "state": {"siid": 2, "piid": 1},
+    "fan_speed": {"siid": 2, "piid": 6},
+    "operating_mode": {"siid": 2, "piid": 4},
+    "mop_state": {"siid": 16, "piid": 1},
+    "water_level": {"siid": 2, "piid": 5},
+    "main_brush_life_level": {"siid": 14, "piid": 1},
+    #  "main_brush_life_time": {"siid": 14, "piid": 2},
+    "side_brush_life_level": {"siid": 15, "piid": 1},
+    #  "side_brush_life_time": {"siid": 15, "piid": 2},
+    "filter_life_level": {"siid": 11, "piid": 1},
+    #  "filter_life_time": {"siid": 11, "piid": 2},
+    "clean_area": {"siid": 9, "piid": 1},
+    "clean_time": {"siid": 18, "piid": 5},
+    "total_clean_count": {"siid": 9, "piid": 5},
+    #   "total_clean_area": {"siid": 9, "piid": 3}, #throws error
+    #  "dnd_enabled": {"siid": 12, "piid": 2},
+    #  "audio_volume": {"siid": 4, "piid": 2},
+    #  "direction_key": {"siid": 8, "piid": 1}
+    "home": {"siid": 2, "aiid": 3},
+    "find": {"siid": 6, "aiid": 1},
+    "start": {"siid": 2, "aiid": 1},
+    "stop": {"siid": 2, "aiid": 2},
+    "reset_main_brush_life_level":  {"siid": 14, "aiid": 1},
+    "reset_side_brush_life_level":  {"siid": 15, "aiid": 1},
+    "reset_filter_life_level":  {"siid": 11, "aiid": 1}
+    }
+}
+
+error_codes = {
+    0: "No error",
+    1: "Left Wheel stuck",
+    2: "Right Wheel stuck",
+    3: "Cliff error",
+    4: "Low battery",
+    5: "Bump error",
+    6: "Main Brush Error",
+    7: "Side Brush Error",
+    8: "Fan Motor Error",
+    9: "Dustbin Error",
+    10: "Charging Error",
+    11: "No Water Error",
+    12: "Pick Up Error"
+}
+
+class G1ChargeState(Enum):
+    Discharging = 0
+    Charging = 1
+    FullyCharged = 2
+
+class G1State(Enum):
+    """Vacuum Status"""
+    Idle = 1
+    Sweeping = 2
+    Paused = 3
+    Error = 4
+    Charging = 5
+    GoCharging = 6
+
+class G1Consumable(Enum):
+    """Consumables"""
+    MainBrush = "main_brush_life_level"
+    SideBrush = "side_brush_life_level"
+    Filter = "filter_life_level"
+
+class G1VacuumMode(Enum):
+    """Vacuum Mode"""
+    GlobalClean = 1
+    SpotClean = 2
+    Wiping = 3
+
+class G1WaterLevel(Enum):
+    """Water Flow Level"""
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+
+class G1FanSpeed(Enum):
+    """Fan speeds, same as for ViomiVacuum."""
+    Mute = 0
+    Standard = 1
+    Medium = 2
+    High = 3
+
+class G1Languages(Enum):
+    Chinese = 0
+    English = 1
+
+class G1MopState(Enum):
+    Off = 0
+    On = 1
+
+
+class G1Status(DeviceStatus):
+    """Container for status reports from the Mijia Vacuum G1."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
```

Comment:
Please add an example response from the device (e.g., by using miiocli in debug mode to get the payload) like shown here: https://github.com/rytilahti/python-miio/blob/master/miio/airconditioner_miot.py#L158